src/main.zig

```diff
@@ -267,11 +270,13 @@ pub fn Deserializer(comptime ReaderType: type) type {
                 format(.bin_32) => try self.reader.readIntBig(u32),
                 else => return error.MismatchingFormatType,
             };
-            if (len > size - self.index) return error.BufferTooSmall;
 
-            const old = self.index;
-            self.index += try self.reader.readAll(self.buffer[self.index .. self.index + len]);
-            return self.buffer[old..self.index];
+            var buffer = try self.allocator.alloc(u8, len);
+            var actual_len = try self.reader.readAll(buffer);
```

Comment:
Those variables can be constant. Same for the other instances.

Reply:
Hi, thanks for the feedback. I did the changes and also fixed the `serializeBin`, to make it consistent with it's `deserializeBin` counterpart. 😃 

I added a test for it since there was none yet. I had to create an individual test for it, instead of simply adding another entry to the `test_cases` variable, because the value input type needs to be `[]u8`, and in the test cases I could only ever get it to `[]const u8` (which makes sense, but would mean it would follow the serializeString instead of the serializeBin path).